src/inference/include/openvino/runtime/properties.hpp

```diff
@@ -0,0 +1,638 @@
+}
+/** @endcond */
+
+/**
+ * @brief High-level OpenVINO model priority hint
+ * Defines what model should be provided with more performant bounded resource first
+ */
+static constexpr Key<ModelPriority> model_priority{"MODEL_PRIORITY"};
+
+/**
+ * @brief Enum to define possible performance mode hints
+ */
+enum class PerformanceMode {
+    LATENCY = 0,
+    THROUGHPUT = 1,
+};
+
+/** @cond INTERNAL */
+inline std::ostream& operator<<(std::ostream& os, const PerformanceMode& performance_mode) {
+    switch (performance_mode) {
+    case PerformanceMode::LATENCY:
+        return os << "LATENCY";
+    case PerformanceMode::THROUGHPUT:
+        return os << "THROUGHPUT";
+    default:
+        throw ov::Exception{"Unsupported performance measure hint"};
+    }
+}
+
+inline std::istream& operator>>(std::istream& is, PerformanceMode& performance_mode) {
+    std::string str;
+    is >> str;
+    if (str == "LATENCY") {
+        performance_mode = PerformanceMode::LATENCY;
+    } else if (str == "THROUGHPUT") {
+        performance_mode = PerformanceMode::THROUGHPUT;
+    } else {
+        throw ov::Exception{"Unsupported performance measure: " + str};
+    }
+    return is;
+}
+/** @endcond */
+
+/**
+ * @brief High-level OpenVINO Performance Hints
+ * unlike low-level config keys that are individual (per-device), the hints are somthing that every device accepts
+ * and turns into device-specific settings
+ */
+static constexpr Key<PerformanceMode> performance_mode{"PERFORMANCE_HINT"};
+
+/**
+ * @brief (Optional) config key that backs the (above) Performance Hints
+ * by giving additional information on how many inference requests the application will be keeping in flight
+ * usually this value comes from the actual use-case (e.g. number of video-cameras, or other sources of inputs)
+ */
+static constexpr Key<uint32_t> num_requests{"PERFORMANCE_HINT_NUM_REQUESTS"};
+
+/**
+ * @brief Enum to define possible performance hints
+ */
+enum class CalculationMode {
+    PERFORMANCE = 0,
+    ACCURACY = 1,
+};
+
+/** @cond INTERNAL */
+inline std::ostream& operator<<(std::ostream& os, const CalculationMode& calculation_mode) {
+    switch (calculation_mode) {
+    case CalculationMode::PERFORMANCE:
+        return os << "PERFORMANCE";
+    case CalculationMode::ACCURACY:
+        return os << "ACCURACY";
+    default:
+        throw ov::Exception{"Unsupported calculation mode hint"};
+    }
+}
+
+inline std::istream& operator>>(std::istream& is, CalculationMode& calculation_mode) {
+    std::string str;
+    is >> str;
+    if (str == "PERFORMANCE") {
+        calculation_mode = CalculationMode::PERFORMANCE;
+    } else if (str == "ACCURACY") {
+        calculation_mode = CalculationMode::ACCURACY;
+    } else {
+        throw ov::Exception{"Unsupported calculation mode: " + str};
+    }
+    return is;
+}
+/** @endcond */
+
+/**
+ * @brief Defines runtime precision mode
+ */
+static constexpr Key<element::Type> precision_mode{"PRECISION_MODE"};
```

Comment:
@vladimir-paramuzov should we introduce this key in current PR? you it's not a part of 2022.1?